akka-typed-tests/src/test/scala/akka/typed/persistence/scaladsl/PersistentActorCompileOnlyTest.scala

```diff
@@ -378,4 +378,29 @@ object PersistentActorCompileOnlyTest {
 
   }
 
+  object Stopping {
+    sealed trait Command
+    case object Enough extends Command
+
+    sealed trait Event
+    case object Done extends Event
+
+    type State = Unit
+
+    PersistentActor.persistent[Command, Event, State](
+      persistenceId = "myPersistenceId",
+      initialState = (),
+      actions = Actions { (cmd, _, _) ⇒
+        cmd match {
+          case Enough ⇒ Persist(Done)
+            .andThen((_: State) ⇒ println("yay"))
+            // Ugh it doesn't make much sense to have to include the type parameters here...
+            .andThen(Stop[Event, State]())
```

Comment:
I couldn't figure out how to make the type inference propagate nicely yet...

Reply:
cop-out approach is of course to introduce `andThenStop` (`def andThenStop: PersistentEffect[Event, State] = andThen(Stop[Event, State]())`), but I wonder if there's a better way

Reply:
does `Nothing` work? then we could have a `val Stop`

could also try `def Stop[Event, State]`

Reply:
by the way, `Event` shouldn't be need in `PersistentEffect`, only `State`

Reply:
I think we do need `Event`, as we want to make sure `Persist` is only called with `Event`s.

It's kind of interesting that no effect seems to need both `Event` and `State` though, only one or the other.